scopesim/effects/fits_headers.py

```diff
@@ -0,0 +1,290 @@
+    - HIERARCH ESO
+     For ESO specific keywords
+    - HIERARCH SIM
+     For ScopeSim specific keywords, like simulation parameters
+    - HIERARCH MIC
+     For MICADO specific keywords, (unsure what these would be yet)
+
+    More HIERARCH style keywords can also be added as needed for other use-cases
+
+    Parameters
+    ----------
+    filename : str, optional
+        Name of a .yaml nested dictionary file. See below for examples
+
+    yaml_string : str, optional
+        A triple-" string containing the contents of a yaml file
+
+    header_dict : nested dicts, optional
+        A series of nested python dictionaries following the format of the
+        examples below. This keyword allows these dicts to be definied directly
+        in the Effect yaml file, rather than in a seperate header keywords file.
+
+
+    Yaml file format
+    ----------------
+    This document is a yaml document.
+    Hence all new keywords should be specified in the form of yaml nested
+    dictionaries.
+    As each ``astropy.HDUList`` contains one or more extensions, the inital level is
+    reserved for a list of keyword groups.
+    For example::
+
+    - ext_type: PrimaryHDU
+      keywords:
+        HIERARCH:
+          ESO:
+            ATM:
+              TEMPERAT: -5
+
+    - ext_number: [1, 2]
+      keywords:
+        HIERARCH:
+          ESO:
+            DET:
+              DIT: [5, '[s] exposure length']   # example of adding a comment
+
+    The keywords can be added to one or more extensions, based on one of the
+    following ``ext_`` qualifiers: ``ext_name``, ``ext_number``, ``ext_type``
+
+    Each of these ``ext_`` qualifiers can be a ``str`` or a ``list``.
+    For a list, ScopeSim will add the keywords to all extensions matching the
+    specified type/name/number
+
+    The above example will result in the following keyword added to:
+
+    - PrimaryHDU (ext 0):
+
+      header['HIERARCH ESO ATM TEMPERAT'] = -5
+
+    - Extensions 1 and 2 (regardless of type):
+
+      header['HIERARCH ESO DET DIT'] = (5, '[s] exposure length')
+
+    Resolved and un-resolved keywords
+    ---------------------------------
+    ScopeSim uses bang-strings to resolve global parameters.
+    E.g: ``from_currsys('!ATMO.temperature')`` will resolve to a float
+    These bang-strings will be resolved automatically in the ``keywords`` dictionary
+    section.
+
+    If the keywords bang-string should instead remain unresolved and the string
+    added verbatim to the header, we use the ``unresolved_keywords`` dictionary
+    section.
+
+    Additionally, new functionality will be added to ScopeSim to resolve the
+    kwargs/meta parameters of Effect objects.
+    The format for this will be to use a new type: the hash-string.
+    This will have this format::
+
+      #<optical_element_name>.<effect_name>.<kwarg_name>
+
+    For example, the temperature of the MICADO detector array can be accessed by::
+
+      '#MICADO_DET.full_detector_array.temperature'
```

Comment:
Why is this `MICADO_DET` and not `micado_detector_array`?

According to the [irdb, the "name of the optical element" is `micado_detector_array`](https://github.com/AstarVienna/irdb/blob/31f7b29bc11fdc6de6a551987cbcf4f0626f2d9e/MICADO/MICADO_H4RG.yaml#L4), so it should be that in the fits header yaml file as well.

(I don't think that is a good name though; it should just be `detector_array`, since it is already in the MICADO namespace, so it obviously is the MICADO one. But that is for a PR in the IRDB.)